crates/metadata/src/tests.rs

```diff
@@ -389,3 +389,35 @@ fn construct_runtime_contract_spec() {
     );
     assert_eq!(event_spec, expected_event_spec);
 }
+
+/// Test for trimming in docs for event specification
+#[test]
+fn trim_events_in_specification() {
+    // GIVEN
+    let label = "foo";
+    let constructor_spec = ConstructorSpec::from_label(label)
+        .selector(123_456u32.to_be_bytes())
+        .docs(vec![" Foo event"])
+        .payable(Default::default())
+        .done();
+    let mut registry = Registry::new();
+    let compact_spec = constructor_spec.into_portable(&mut registry);
+
+    // WHEN
+    let json = serde_json::to_value(&compact_spec).unwrap();
+    let deserialized: ConstructorSpec<PortableForm> =
```

Comment:
Should be testing the `EventSpecBuilder` and `EventParamSpecBuilder` which you have changed.